common/src/main/scala/explore/components/graphql/SubscriptionRender.scala

```diff
@@ -0,0 +1,89 @@
+// Copyright (c) 2016-2020 Association of Universities for Research in Astronomy, Inc. (AURA)
+// For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+package explore.components.graphql
+
+import japgolly.scalajs.react._
+import japgolly.scalajs.react.vdom.html_<^._
+import react.common._
+import react.semanticui.elements.icon.Icon
+import react.semanticui.sizes._
+import clue.GraphQLStreamingClient
+import cats.implicits._
+import cats.effect.IO
+import crystal._
+import crystal.react.StreamRenderer
+import crystal.react.io.implicits._
+import diode.data._
+import diode.react.ReactPot._
+import cats.effect.ContextShift
+
+final case class SubscriptionRender[D, A](
+  subscribe:      IO[GraphQLStreamingClient[IO]#Subscription[D]],
+  streamModifier: fs2.Stream[IO, D] => fs2.Stream[IO, A] = identity[fs2.Stream[IO, D]] _
+)(
+  val valueRender: A => VdomNode,
+  val onNewData:   IO[Unit] = IO.unit
+)(
+  implicit val cs: ContextShift[IO]
```

Comment:
Do we need it implicit here? we are not going to use it inside props or do we?

Reply:
No, but having it as implicit also means the caller doesn't need to specify it if it has an implicit instance in scope.